kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/Tls.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault.config;
```

Comment:
I was imagining that the Tls class (and friends) would move to kroxylicious-api avoiding the need to duplicate the class.  It would mean that any filter and any kms implementation could re-use.

Reply:
I've moved the configuration classes into kroxylicious-api. This time I've put no logic about `TrustManager`s into the config classes, the only bit of smarts is password file classes know how to load the password.

The two clients (kroxylicious-runtime and the vault KMS) use a visitor to get at the concrete TrustProvider type and then each have their own logic for constructing SSL contexts.

I also wondered about using a sealed hierarchy and switching but we need a higher lang level to support the pattern matching switch.

Reply:
if using Visitor is a bit overwrought, happy enough to switch to instanceof, what do you all prefer?